lib/spack/spack/stage.py

```diff
@@ -39,35 +38,44 @@
 
 def _create_stage_root(path):
     """Create the stage root directory and ensure appropriate access perms."""
-    assert path.startswith(os.path.sep) and len(path.strip()) > 1
+    assert path.startswith(os.sep) and len(path.strip()) > 1
 
-    curr_dir = os.path.sep
+    err_msg = 'Cannot create stage root {0}: Access to {1} is denied'
+
+    user_uid = os.getuid()
     parts = path.strip(os.path.sep).split(os.path.sep)
-    for i, part in enumerate(parts):
-        curr_dir = os.path.join(curr_dir, part)
-        if not os.path.exists(curr_dir):
-            rest = os.path.join(os.path.sep, *parts[i + 1:])
-            user_parts = rest.partition(os.path.sep + getpass.getuser())
-            if len(user_parts[0]) > 0:
-                # Ensure access controls of subdirs created above `$user`
-                # inherit from the parent and share the group.
-                stats = os.stat(os.path.dirname(curr_dir))
-                curr_dir = ''.join([curr_dir, user_parts[0]])
-                mkdirp(curr_dir, group=stats.st_gid, mode=stats.st_mode)
-
-            user_subdirs = ''.join(user_parts[1:])
-            if len(user_subdirs) > 0:
+
+    i = 1 if path[0] == os.sep else 0
```

Comment:
Great catch.  Thanks.  (One problem with evolving functionality -- here it's change to a more limited scope -- is you don't always "see" these things.)

Reply:
Why should `llnl.util.filesystem.prefixes` should remove the starting `os.sep`?  

That function returns a list of ancestor paths.

Reply:
BTW.  This approach was influenced by the comment (https://github.com/spack/spack/pull/12516#discussion_r320559905) suggesting it would be cleaner than what I had before.

Reply:
> Why should llnl.util.filesystem.prefixes remove the starting os.sep?

I imagine all users of the function would not want to use the initial `''` entry that appears for an absolute path - that it exists seems more a side-effect behavior of `os.split` than a desirable result when examining a path string.

Reply:
Refactored.